sdks/python/apache_beam/runners/worker/data_plane.py

```diff
@@ -69,22 +69,78 @@ def close(self):
       self._close_callback(self.get())
 
 
-class DataChannel(with_metaclass(abc.ABCMeta, object)):
-  """Represents a channel for reading and writing data over the data plane.
+class DataChannel(object):
+
+  def __init__(self):
+    self.data_conn = DataChannelConnection()
+
+
+class InMemoryDataChannel(DataChannel):
+  """An in-memory implementation of a DataChannel.
+
+  This channel is two-sided.  What is written to one side is read by the other.
+  The inverse() method returns the other side of an instance.
+  """
+
+  def __init__(self, inverse=None, data_conn=None):
+    self.data_conn = data_conn or InMemoryDataChannelConnection()
+    self._inverse = inverse or InMemoryDataChannel(
+        self, self.data_conn.inverse())
+
+  def inverse(self):
+    return self._inverse
 
-  Read from this channel with the input_elements method::
+
+class GrpcClientDataChannel(DataChannel):
+  """A DataChannel wrapping the client side of a BeamFnData connection."""
+
+  def __init__(self, data_stub):
+    self.data_conn = GrpcDataChannelConnection()
+    self.data_conn._start_reader(data_stub.Data(
+        self.data_conn._write_outputs()))
+
+
+class GrpcServerDataChannel(
+    beam_fn_api_pb2_grpc.BeamFnDataServicer, DataChannel):
+  """A DataChannel wrapping the server side of a BeamFnData connection."""
+
+  def __init__(self):
+    self.data_conn = GrpcDataChannelConnection()
+
+  def Data(self, elements_iterator, context):
+    # py27 doesn't support recursive import at module level
```

Comment:
circular imports work fine with py3, at least with py36. I haven't tested with py35 and py37.
But we don't need it any more after creating `connections_by_worker_id` at channel rather than at runner side.